src/rendergraph/common/rendergraph/attributeset.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "backend/attributeset.h"
+
+namespace rendergraph {
+class AttributeSet;
+}
+
+class rendergraph::AttributeSet : public rendergraph::backend::AttributeSet {
```

Comment:
* It makes it much less likely that you forget to closing '}' (and would even more so if things were properly indented)
* It allows you to not have an indentation rule exception for namespace. Not indenting namespace is a kludge because having the extra indentation for your classes is annoying. I am very much against this exception in the clang-format rules, I get very nervous when I see a '{' '}' block without the content indented, it is evil!
* You see at a glance in which namespace a class lives.

Reply:
I am very much used to this pattern, coming from a code base with clang format rules that *does* indent the name space and that uses nested namespaces a lot. I also used it in the allshader code so it already forms part of Mixxx :-)

If this really bothers you I can remove it, but I don't see any drawback apart from causing some minor confusion when you first see this pattern - and hopefully subsequent admiration for it's elegance, compared to the kludge of not indenting namespaces ;-)

Reply:
No it doesn't bother me particularly. I'm just not used to it and the duplication it causes seems suboptimal. I'm open to trying this for now.